File: api/src/main/java/com/sahilkumar/api/expense/Expense.java
package com.sahilkumar.api.expense;

import com.sahilkumar.api.common.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import java.time.LocalDate;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "expenses")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Expense extends BaseEntity {

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 40)
    private ExpenseCategory category;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal amount;

    @Column(length = 500)
    private String notes;

    @Column(name = "expense_date", nullable = false)
    private LocalDate expenseDate;
}

File: android/app/src/main/java/com/sahilkumar/dfms/model/Expense.kt
package com.sahilkumar.dfms.model

import java.math.BigDecimal

data class ExpenseResponse(
    val id: String,
    val category: ExpenseCategory,
    val amount: BigDecimal,
    val notes: String?,
    val expenseDate: String,
    val cowId: String?,
    val cowTagNo: String?,
)

data class ExpenseRequest(
    val category: ExpenseCategory,
    val amount: BigDecimal,
    val notes: String?,
    val expenseDate: String,
    val cowId: String?,
)
